### PR TITLE
Add documentation for @StreamListener dispatching

### LINKED ISSUE #820: Add documentation for @StreamListener with conditions

### PR DESCRIPTION
**Pull request by mbogoevici**
Fixes #820

### CODE REVIEW

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 1]
```diff
@@ -453,9 +453,7 @@ In the case of `@StreamListener`, the `MessageConverter` mechanism will use the
 As with other Spring Messaging methods, method arguments can be annotated with `@Payload`, `@Headers` and `@Header`.
 
 [NOTE]
-====
 For methods which return data, you must use the `@SendTo` annotation to specify the output binding destination for data returned by the method:
-====
```
Comment (garyrussell) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Why did you remove these? `NOTE`s have 2 syntaxes

Admonition paragraph
```
NOTE:foobar
```
and 

Admonition Block
```
[NOTE]
====
foobar
====
```

See http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#admon-bl

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 2]
```diff
@@ -472,7 +470,43 @@ public class TransformProcessor {
...
+
+===== Using @StreamListener for dispatching messages to multiple methods
+
+Since version 1.2, Spring Cloud Stream supports dispatching messages to multiple `@StreamListener` methods registered on an input channel, based on a condition.
+
+In order to be eligible to support conditional dispatching, a method must satisfy the follow conditions:
+
+* the method must not return a value;
+* to be an individual message handling method (reactive API methods are not supported);
```
Comment (garyrussell) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: `s/the method/it/`
`s/to/it must/`

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 3]
```diff
@@ -472,7 +470,43 @@ public class TransformProcessor {
...
+    @StreamListener(target = Sink.INPUT, condition = "headers['type']=='bar'")
+    public void receiveBar(@Payload BarPojo barPojo) {
+       // handle the message
+    }
+}
+----
+
+
+[NOTE]
+Dispatching via `@StreamListener` conditions is only supported for handlers of individual messages, and not for reactive programming support (described below).
```
Comment (garyrussell) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Needs a block
```
====

====
```

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc (hunk 1 above)
Comment (mbogoevici) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Interesting .. `[NOTE]` seems to work directly the paragraph too. Probably shouldn't stretch a loophole :). Actually I was trying to fix an issue where we had an extra block closing element which threw off the docs.

**Comment by garyrussell** [comment]
Merged as 02fdd7c9ddfcbd3420d2e55f89b47cb3ee15ac3a
